File: lib/models/ble_uuid_type.dart
enum BleUuidType {
  // 通用
  common,
  // 大数据
  largeData,
  // 流式数据
  streaming,
  // OTA
  ota,
}

extension BleUuidTypeExt on BleUuidType {
  static BleUuidType label(String label) {
    switch (label) {
      case "largeData":
        return BleUuidType.largeData;
      case "streaming":
        return BleUuidType.streaming;
      case "ota":
        return BleUuidType.ota;
      default:
        return BleUuidType.common;
    }
  }

  bool get isCommon => this == BleUuidType.common;
  bool get isLargeData => this == BleUuidType.largeData;
  bool get isStreaming => this == BleUuidType.streaming;
  bool get isOta => this == BleUuidType.ota;
}

File: ios/Classes/ble/models/BleUuidType.swift
//
//  BleUuidType.swift
//  flutter_ezw_ble
//
//  Created by Whiskee on 2025/4/23.
//

enum BleUuidType: String, Codable {
    case common
    case largeData
    case streaming
    case ota
    
    var isCommon: Bool {
        get {
            return self == .common
        }
    }
    
    var isLargeData: Bool {
        get {
            return self == .largeData
        }
    }
    
    var isStreaming: Bool {
        get {
            return self == .streaming
        }
    }
    
    var isOta: Bool {
        get {
            return self == .ota
        }
    }
}

